### PR TITLE
Add Careers processor/mapping/query

### PR DESCRIPTION
**Pull request by kurtw**
This is the processor that takes in the Jobs API data and uses it for the Careers section. 

@dpford @jimmynotjim

### CODE REVIEW

File: src/_queries/careers.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "Careers",
+  "query": {
+    "doc_type": "career",
+    "size": 100,
+    "sort": "title"
```
Comment (jimmynotjim) [inline src/_queries/careers.json]: Shouldn't sort be date?

File: src/_settings/mappings/career.json [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      "properties": {
+        "announcement_number": {
+          "type": "string"
+        },
+        "application_type": {
+          "properties": {
+            "active": {
+              "type": "boolean"
+            },
+            "applicant_type": {
```
Comment (jimmynotjim) [inline src/_settings/mappings/career.json]: `applicant_type.applicant_type` seems odd way to structure the data. Should we change this to `name` or `title`?

File: src/_settings/mappings/career.json [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    "active": {
+      "type": "boolean"
+    },
+    "applicant_types": {
+      "properties": {
+        "announcement_number": {
+          "type": "string"
+        },
+        "application_type": {
+          "properties": {
```
Comment (jimmynotjim) [inline src/_settings/mappings/career.json]: Can we remove the `properties` object and access the `active` boolean right off of `application_type`?

File: src/_settings/mappings/career.json [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            },
+            "slug": {
+              "type": "string"
+            }
+          }
+        },
+        "id": {
+          "type": "long"
+        },
+        "is_usajobs": {
```
Comment (jimmynotjim) [inline src/_settings/mappings/career.json]: Are there postings that aren't USAJobs? We'll need to swap out some content in the page if so.

File: src/_settings/mappings/career.json (hunk 3 above)
Comment (kurtw) [inline src/_settings/mappings/career.json]: So properties` is just something in the mapping that tells elasticsearch this is an object with the fields as it's members.

File: src/_settings/mappings/career.json (hunk 4 above)
Comment (kurtw) [inline src/_settings/mappings/career.json]: Not sure. We can run a filter and see what comes out.

File: src/_queries/careers.json (hunk 1 above)
Comment (kurtw) [inline src/_queries/careers.json]: I could see using `close_date`. Also, if we really don't care about past jobs then we should just filter it by the `active=true` or something.

File: src/_settings/mappings/career.json (hunk 2 above)
Comment (kurtw) [inline src/_settings/mappings/career.json]: We could change it to either, but you'd still need to access it like `applicant_types.application_type.name`

**Comment by rosskarchner** [comment]
The API is live: http://www.consumerfinance.gov/jobs/jobs.json

File: src/_settings/mappings/career.json (hunk 3 above)
Comment (jimmynotjim) [inline src/_settings/mappings/career.json]: Ah ok, thanks for the explainer

File: src/_settings/mappings/career.json (hunk 2 above)
Comment (jimmynotjim) [inline src/_settings/mappings/career.json]: You would, but then it's a bit clearer exactly what you're looking at

**Comment by kurtw** [comment]
Updated with suggestions. Now the search query only returns jobs with the close date on and after the current date, and the `applicant_types.application_type.applicant_type` is now accessible via `applicant_types.application_type.name`.

**Comment by jimmynotjim** [comment]
:+1:

**Comment by dpford** [comment]
:clapper:

**Pull request merged**